packages/field/src/FormRegistrationMixins.js

```diff
@@ -0,0 +1,77 @@
+import { dedupeMixin } from '@lion/core';
+
+/**
+ * #RegisterFormElementMixin :
+ *
+ * This Mixin registers form elements, it's applied on:
+ * - FormControlMixin (which is extended to LionField, LionInput, LionTextarea, LionSelect etc. etc.)
+ *
+ * @polymerMixin
+ * @mixinFunction
+ */
+export const FormRegisteringMixin = dedupeMixin(
+  superclass =>
+    // eslint-disable-next-line no-shadow, no-unused-vars
+    class FormRegisteringMixin extends superclass {
+      firstUpdated(c) {
+        super.firstUpdated(c);
+        this.__setupRegistrationHook();
+      }
+
+      __setupRegistrationHook() {
+        if (document.body.__lionFormRegistrationReady) {
+          this._registerFormElement();
+        } else {
+          document.body.addEventListener('lion-form-registration-ready', () => {
```

Comment:
I think this creates a problem if I have more than 1 lion-form on the page. Can we add a unique identifiers to it to support multiple forms?